src/rabbit_peer_discovery.erl

```diff
@@ -99,6 +103,56 @@ maybe_unregister() ->
   end.
 
 
+-spec maybe_inject_randomized_delay() -> ok.
+maybe_inject_randomized_delay() ->
+  Backend = backend(),
+  case Backend:supports_registration() of
```

Comment:
Should we add `supports_registration()` to [rabbit_peer_discovery_backend](
https://github.com/rabbitmq/rabbitmq-common/blob/master/src/rabbit_peer_discovery_backend.erl) ?